### DISCUSSION TITLE
Laravel 8: File storage issue. store($file) saves, download($table->file) does not find

### DISCUSSION THREAD

**Discussion by alsuara** [laravel/framework#38591] [Q&A]
- Laravel Version: 8.50.0
- PHP Version: 7.3.20
- Database Driver & Version: 10.4.13-MariaDB

### Description:

According to the Laravel's file storage documentation ( https://laravel.com/docs/8.x/filesystem ) I am performing a `store()` operation to save an uploaded file in my local disk, but then, when I use `download()` to have it back, I receive a «file not found error».

Perhaps I am mistaken nd someone can tell me what I am doing wrong, but for the case, I preferred to rise the issue.

Thanks and best regard.

### Steps To Reproduce:

I am trying to upload a pdf to my local storage doing: `$validated['pdf'] = $request->pdf->store('');` and then `$model->update($validated);` This stores a random-name with «pdf» extension file in the database (for example **jf9T0Oo9OK7EshXjsxjNghKHczShLWHa4mwYCMg7.pdf** ).

Then I try to donwload the file doing `return response()->download($model->pdf);` but I receive an error:

**Symfony\Component\HttpFoundation\File\Exception\FileNotFoundException
The file "71NgZS0q3TMzbe8eSHkH36kfPIFxLrnbqNnmMyNh.pdf" does not exist**

In order to understand what is happening, I have tried to do it using storage_path() and I could see that in the path, the directory «app» under «storage» is missing, so I could make the donwload with the following turnaround:

`return response()->download(storage_path('app/' . $model->pdf));`

(Am am working in a local installation of XAMPP for OS X 7.3.20)

**Comment by NickSdot** [comment]
Try `$path = $request->file('fileInRequest')->store('pathToSaveFileTo');`. You can `dd($request->file());` to check how to access the file.

Maybe better to close this issue and instead start a discussion here:
https://github.com/laravel/framework/discussions

**Reply by alsuara** [reply]
With your version, the same problem persists (when tried to be downloaded, the uploaded file it's not found) as the problem doesn't seem to be in the `store()` method, but in the `file()` and `download()` methods, so it needs the same turnaround (using `storage_path()`) to download the file.

The storage is correctly done as documented in **https://laravel.com/docs/8.x/filesystem#file-uploads**, but `storage_path()` is not mentioned in **https://laravel.com/docs/8.x/filesystem#downloading-files** neither in **https://laravel.com/docs/8.x/responses#file-responses**. 

It seems that at least, a documentation issue persists.
